src/modules/rlm_imap/imap.h

```diff
@@ -0,0 +1,225 @@
+} http_body_type_t;
+
+typedef enum {
+    imap_HTTP_AUTH_UNKNOWN = 0,
+    imap_HTTP_AUTH_NONE,
+    imap_HTTP_AUTH_TLS_SRP,
+    imap_HTTP_AUTH_BASIC,
+    imap_HTTP_AUTH_DIGEST,
+    imap_HTTP_AUTH_DIGEST_IE,
+    imap_HTTP_AUTH_GSSNEGOTIATE,
+    imap_HTTP_AUTH_NTLM,
+    imap_HTTP_AUTH_NTLM_WB,
+    imap_HTTP_AUTH_ANY,
+    imap_HTTP_AUTH_ANY_SAFE,
+    imap_HTTP_AUTH_NUM_ENTRIES
+} http_auth_type_t;
+
+//I feel like a lot of this might be bloat
+typedef struct {
+	char const        		*name;        //!< Section name.
+	char const        		*uri;        //!< URI to send HTTP request to.
+
+	char const        		*proxy;        //!< Send request via this proxy.
+
+	char const        		*method_str;    //!< The string version of the HTTP method.
+	http_method_t        		method;        //!< What HTTP method should be used, GET, POST etc...
+
+	char const        		*body_str;    //!< The string version of the encoding/content type.
+	http_body_type_t    		body;        //!< What encoding type should be used.
+
+	char const        		*force_to_str;    //!< Force decoding with this decoder.
+	http_body_type_t    		force_to;    //!< Override the Content-Type header in the response
+                        //!< to force decoding as a particular type.
+
+	char const        		*data;        //!< Custom body data (optional).
+
+	bool            		auth_is_set;    //!< Whether a value was provided for auth_str.
+
+	http_auth_type_t    		auth;        //!< HTTP auth type.
+
+	bool            		require_auth;    //!< Whether HTTP-Auth is required or not.
+	char const        		*username;    //!< Username used for HTTP-Auth
+	char const        		*password;    //!< Password used for HTTP-Auth
+
+	char const        		*tls_certificate_file;
+	char const        		*tls_private_key_file;
+	char const        		*tls_private_key_password;
+	char const        		*tls_ca_file;
+	char const        		*tls_ca_issuer_file;
+	char const        		*tls_ca_path;
+	char const        		*tls_random_file;
+	bool            		tls_check_cert;
+	bool            		tls_check_cert_cn;
+	bool            		tls_extract_cert_attrs;
+
+	fr_time_delta_t        		timeout;    //!< Timeout timeval.
+	uint32_t        		chunk;        //!< Max chunk-size (mainly for testing the encoders)
+	size_t            		max_body_in;    //!< Maximum size of incoming data.
+} rlm_imap_section_t;
+
+
+
+typedef enum {
+	READ_STATE_INIT    = 0,
+	READ_STATE_ATTR_BEGIN,
+	READ_STATE_ATTR_CONT,
+	READ_STATE_END,
+} read_state_t;
+
+/*
+ *    States for the response parser
+ */
+typedef enum {
+	WRITE_STATE_INIT = 0,
+	WRITE_STATE_PARSE_HEADERS,
+	WRITE_STATE_PARSE_CONTENT,
+	WRITE_STATE_DISCARD,
+} write_state_t;
+
+
+typedef struct {
+    /*so, for this module it'd be where you'd add things like the IMAP server's address
+    *configuration options that specify whether to use TLS
+    *what certificates to use to validate the one the IMAP server presents etc...
+     */
+	bool        			boolean;
```

Comment:
These configuration items are unused, and can be deleted.

See the `rlm_ldap` for some examples of a `server` configuration and `tls` configuration section.